### PR TITLE
Grw-1702 / feat / PDP: Navigation and Tabs menu sticky together

### PR DESCRIPTION
**Pull request by KarnellSchultz**
## Describe your changes

Add the ability for the nav and the tab menu on PDP to scroll nicely with one another. 

The below video shows the scroll. 

https://user-images.githubusercontent.com/50870173/200324103-df8968bb-b1dd-4072-bae9-adf191371136.mov

This is how you'd expect two sticky position elements to scroll with one another. With one element pushing the other off of the page rather than going behind it. 

https://hedvig-dot-jregh850j-hedvig.vercel.app/se/products/bil

### The reason this is not done with css
For position sticky to work with these two components they'd need to have the same parent container. To do that we'd need to make major changes to our application's markup. Making those changes would come with other issues that are really not worth it / not acceptable. 

## Checklist before requesting a review

- [x] I have performed a self-review of my code

[GRW-1702]: https://hedvig.atlassian.net/browse/GRW-1702?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/store/src/blocks/ImageBlock.tsx [hunk 1]
```diff
@@ -32,6 +32,7 @@ const Wrapper = styled.div<{ margins: boolean }>(({ theme, margins = true }) =>
   paddingLeft: margins ? theme.space[4] : 0,
   paddingRight: margins ? theme.space[4] : 0,
   position: 'relative',
+  zIndex: -1,
```
Comment (gustaveen) [inline apps/store/src/blocks/ImageBlock.tsx:35]: Should we isolate the zIndex settings to the Tablists instead? Just thinking the ImageBlock or other blocks shouldn't have to know about this feature :)

File: apps/store/src/components/TopMenu/useTopMenuStickyScroll.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  return scrollPosition
+}
+
+export const useTopMenuStickyScroll = () => {
+  const [state, setState] = useState({ distanceFromTop: 0, elementHeight: 0 })
+  const scrollPosition = useScrollPosition()
+
+  useEffect(() => {
+    const tabListElement = document?.getElementById('tab-list')
+    const rect = tabListElement?.getBoundingClientRect()
```
Comment (gustaveen) [inline apps/store/src/components/TopMenu/useTopMenuStickyScroll.tsx:23]: I was a bit curious if we can use the [Intersection Observer API](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API) for this to make it a bit cheaper on the main thread. But from what I can tell it isn't the optimal use case to update a pixel value on an element like that. Did you find any examples using that API @KarnellSchultz :) 

The current implementation works really nice when testing! ⭐️

Comment (KarnellSchultz) [reply apps/store/src/components/TopMenu/useTopMenuStickyScroll.tsx:23]: @gustaveen  Good call, after looking into Intersection observer that seems to be the way to go. Found some interesting things [about performance](https://itnext.io/1v1-scroll-listener-vs-intersection-observers-469a26ab9eb6) too.

I'll make the swap. 

As for incrementing the pixel. I can't think of another way to move the nav off the view port 🤔

File: apps/store/src/blocks/ImageBlock.tsx (hunk 1 above)
Comment (KarnellSchultz) [reply apps/store/src/blocks/ImageBlock.tsx:35]: Without this line the images were in front of the Tablist when scrolling. But let's see if we can sort this out another way.

**Pull request closed without merging**